### DISCUSSION TITLE
How to properly handle "scenes"/change the state of what content should be displayed.

### DISCUSSION THREAD

**Discussion by JustASpeedrunner** [emilk/egui#3361] [Q&A] [answered]
As title, making an app that, once a certain action is performed, will go onto another "scene" with different content, relating to the action the user input. I searched around a bit and couldn't find anything, but I'm new so I could just be ignorant to the solution.

**Comment by YgorSouza** [comment, marked as answer]
egui redraws the whole UI on every frame, and it will draw whatever you tell it to draw, so you can create a variable to keep track of the current scene you are in, then update it according to the actions the user took, then have the logic to decide which UI to draw according to the state of that variable. I.e., a finite state machine.

The [plot demo](https://github.com/emilk/egui/blob/08c46acba131f131438ef80f63b5338021aa325d/crates/egui_demo_lib/src/demo/plot_demo.rs#L81-L118) from the [demo app](https://www.egui.rs/) is a simple example. It has the `open_panel` enum field that can be changed by clicking on the tab headers at the top, and then it matches on the enum to decide which of the demo UIs to draw on that frame.

**Reply by JustASpeedrunner** [reply]
Thank you, this was my initial plan but I wanted to make sure prior to doing so that it was the correct way to do it and that I wasn't missing a built in feature for this.
